the-collab-lab/tcl-42-smart-shopping-list
# Issue 2  - Add Navigation

## Description

We updated the App component to use the  BrowserRouter, Routes and Route from the React-Router-Dom Library in order for this component to act as a switch board for our application.  We created a navigation component which uses the navLink component from React-Router-Dom to display our links with styling on the active link. These links point towards separate components that render either the List view or the Add Item view. 

LEARNINGS: 
We learned to use React-Router-Dom as a tool for site navigation and the navLink component as a simple and effective way to style the active links. 


## Related Issue
closes 2

## Acceptance Criteria

- [✓ ] react-router-dom has been added as a project dependency
- [ ✓] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- [✓ ] When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
- [ ✓] Whichever view is selected, the corresponding link should display in bold text



## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|   ✓ | :100: Add tests            |
|   ✓ | :link: Update dependencies |



### After

![Screen Shot 2022-04-06 at 11 35 18 PM](https://user-images.githubusercontent.com/85757379/162115289-17ca43d1-5c3f-4b78-acc5-63edbf26defb.jpeg)



## Testing Steps / QA Criteria

- Pull down the branch using `git pull origin mp-tt-link-navigation`.
- `git checkout mp-tt-link-navigation` to switch branches.
- `npm install` to update dependencies
- `npm run test` to run tests on components.
- `npm run start` to open a local copy of the app.
- Click the links to see the active link style changes.
